### PR TITLE
[Paid Newletters] tests for paid newsletters regressions

### PR DESCRIPTION
**Pull request by millerf**
Adding regression tests for  https://github.com/Automattic/jetpack/pull/28912 and https://github.com/Automattic/jetpack/pull/28894

## Proposed changes:

* Added PHPunit test

### Other information:

- [X] Have you written new tests for your changes, if applicable?
- [X] Have you checked the E2E test CI results, and verified that your changes do not break them?

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
* Run docker `jetpack docker up -d`

#### Regression tests for  https://github.com/Automattic/jetpack/pull/28912

* Run PHP unit testsuite `jetpack docker phpunit -- -c tests/php.multisite.xml --testsuite=premium-content`
* You can test the regression by reverting 61f3b3604adfd29401b85958656d02c5c5270bc1 and testing 
```
$> git revert 61f3b3604adfd29401b85958656d02c5c5270bc1
[ TYPE ":qa!" to commit the revert ]
$> jetpack docker phpunit -- -c tests/php.multisite.xml --testsuite=premium-content
[ WIL FAILL ]
```

#### Regression tests for  https://github.com/Automattic/jetpack/pull/28894

* Run PHP unit testsuite  `jetpack docker phpunit --  -c tests/php.multisite.xml --testsuite=subscriptions --filter="test_comments_*"`
* You can test the regression by reverting e2c3b99c39047a15de02ca82f23893185916e2d9 and testing
```
$> git revert e2c3b99c39047a15de02ca82f23893185916e2d9
[ TYPE ":qa!" to commit the revert ]
$> jetpack docker phpunit --  -c tests/php.multisite.xml --testsuite=subscriptions --filter="test_comments_*"
[ WILL FAIL ]
```

### CODE REVIEW

File: projects/plugins/jetpack/tests/php/modules/premium-content/test_class.jetpack-premium-content.php [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+			)
+		);
+
+		// We create a plan
+		$plan_id = $this->factory->post->create(
+			array(
+				'post_type' => Jetpack_Memberships::$post_type_plan,
+			)
+		);
+		update_post_meta( $plan_id, 'jetpack_memberships_product_id', $this->product_id );
```
Comment (artpi) [inline projects/plugins/jetpack/tests/php/modules/premium-content/test_class.jetpack-premium-content.php]: Where is this meta used?

**Review by artpi — DISMISSED** [review]
This is ok, however we are missing a test

- subscriber is a paid subscriber
- `comments_open` default is false
- Filter should default to false

File: projects/plugins/jetpack/tests/php/modules/premium-content/test_class.jetpack-premium-content.php (hunk 1 above)
Comment (millerf) [reply projects/plugins/jetpack/tests/php/modules/premium-content/test_class.jetpack-premium-content.php]: copy/paste issue...

Comment (millerf) [reply projects/plugins/jetpack/tests/php/modules/premium-content/test_class.jetpack-premium-content.php]: @artpi this is actually used in https://github.com/Automattic/jetpack/blob/94ddae8284758db97529fad30752b9e11a556bb1/projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php#L237

Stack call:
```
- class-token-subscription-service.php:238, Automattic\Jetpack\Extensions\Premium_Content\Subscription_Service\Token_Subscription_Service::validate_subscriptions()
- class-token-subscription-service.php:94, Automattic\Jetpack\Extensions\Premium_Content\Subscription_Service\Token_Subscription_Service->visitor_can_view_content()
- access-check.php:99, Automattic\Jetpack\Extensions\Premium_Content\current_visitor_can_access()
- test_class.jetpack-premium-content.php:152, WP_Test_Jetpack_Premium_Content->test_access_check_current_visitor_can_access_regular_users()
TestCase.php:1548
```

**Comment by millerf** [comment]
> This is ok, however we are missing a test
> 
> * subscriber is a paid subscriber
> * `comments_open` default is false
> * Filter should default to false

Test was added 👍

File: projects/plugins/jetpack/tests/php/modules/premium-content/test_class.jetpack-premium-content.php (hunk 1 above)
Comment (artpi) [reply projects/plugins/jetpack/tests/php/modules/premium-content/test_class.jetpack-premium-content.php]: Ah I remember now

**Review by artpi — DISMISSED** [review]
Nice work!

**Review by jeherve — CHANGES_REQUESTED** [review]
Thanks for adding tests!

For our futureselves, I wonder if that new `modules/premium-content` directory may prove confusing.
When adding new files and directories to `projects/plugins/jetpack/tests/php/`, we try to follow the file structure one may find in `projects/plugins/jetpack`. This way, when a new contributor (or an old one who hasn't looked at that part of the codebase in a while) wants to check on tests for a specific class, function, or method, they can check the matching file in the tests directory.
Do you think you could move those tests to where they match the different methods we're testing here?

One more thing that can help is to add `@covers` in the docblock of each test, for the same reason.

Thank you!

**Review by jeherve — APPROVED** [review]

**Comment by millerf** [comment]
[THOSE COMMENT DID NOT SEND - ADDING THEM NOW FTR]

> Do you think you could move those tests to where they match the different methods we're testing here?

Sorry I didn't see the `/extensions` folder. Now it should be fine. Do you think I should put it in a sub `_inc/` folder as well?

> One more thing that can help is to add @covers in the docblock of each test, for the same reason.

Didn't know about those. Done 👍

**Pull request merged**